commcare_connect/reports/helpers.py

```diff
@@ -0,0 +1,103 @@
+import calendar
+from collections import defaultdict
+from datetime import datetime
+
+from django.db.models import Max, Q, Sum
+from django.utils.timezone import make_aware
+
+from commcare_connect.opportunity.models import CompletedWork, CompletedWorkStatus, Payment
+
+
+def get_table_data_for_year_month(
+    year=None,
+    month=None,
+    delivery_type=None,
+    group_by_delivery_type=False,
+    program=None,
+    network_manager=None,
+    opportunity=None,
+):
+    if year is None:
+        year = datetime.now().year
+    filter_kwargs = {}
+    if delivery_type:
+        filter_kwargs.update({"opportunity_access__opportunity__delivery_type__slug": delivery_type})
+    if program:
+        filter_kwargs.update({"opportunity_access__opportunity__managedopportunity__program": program})
+    if network_manager:
+        filter_kwargs.update({"opportunity_access__opportunity__organization": network_manager})
+    if opportunity:
+        filter_kwargs.update({"opportunity_access__opportunity": opportunity})
+
+    _, month_end = calendar.monthrange(year, month or 1)
+    start_date = make_aware(datetime(year, month or 1, 1))
+    end_date = make_aware(datetime(year, month or 12, month_end))
+    data = []
+
+    user_set = defaultdict(set)
+    beneficiary_set = defaultdict(set)
+    service_count = defaultdict(int)
+
+    visit_data = (
+        CompletedWork.objects.annotate(work_date=Max("uservisit__visit_date"))
+        .filter(
+            Q(status_modified_date__gte=start_date, status_modified_date__lt=end_date)
+            | Q(status_modified_date__isnull=True, work_date__gte=start_date, work_date__lt=end_date),
+            **filter_kwargs,
```

Comment:
I think we only want to count nonzero dollar deliveries based on the doc

Reply:
Added. [9b88a0f](https://github.com/dimagi/commcare-connect/pull/485/commits/9b88a0fa3367e0cad01254b3b9c951bbb0d28dbb)